asmcomp/asmpackager.ml

```diff
@@ -91,10 +92,33 @@ let make_package_object ppf members targetobj targetname coercion =
         | PM_intf -> None
         | PM_impl _ -> Some(Ident.create_persistent m.pm_name))
       members in
-  Asmgen.compile_implementation ~source_provenance:(Timings.Pack targetname)
-    (chop_extension_if_any objtemp) ppf
-    (Translmod.transl_store_package
-       components (Ident.create_persistent targetname) coercion);
+  let module_ident = Ident.create_persistent targetname in
+  let source_provenance = Timings.Pack targetname in
+  let prefixname = chop_extension_if_any objtemp in
+  if Config.flambda then begin
+    let size, lam =
+      Translmod.transl_package_flambda
+        components module_ident coercion
+    in
+    let flam =
+      Middle_end.middle_end ppf
+        ~source_provenance
+        ~prefixname
+        ~backend
+        ~size
+        ~module_ident
+        ~module_initializer:lam
+    in
+    Asmgen.compile_implementation ~source_provenance
+      prefixname ~backend Asmgen.Flambda ppf flam;
+  end else begin
+    let main_module_block_size, code =
```

Comment:
This suggest to move the definition of lambda_program to Translmod and have transl_store_package return it directly.  Or just keep the old type for Asmgen.compile_implementation (for the version specialized on non-flambda mode).